pkg/cloud/services/eks/identity_provider.go

```diff
@@ -49,14 +49,20 @@ func (s *Service) reconcileIdentityProvider(ctx context.Context) error {
 	}
 
 	s.scope.V(2).Info("creating oidc provider plan", "desired", desired, "current", current)
+	procedures, err := identityprovider.
+		NewPlan(clusterName, current, desired, s.EKSClient, s.scope.Logger).
+		Create(ctx)
 
-	identityProviderPlan := identityprovider.NewPlan(clusterName, current, desired, s.EKSClient, s.scope.Logger)
-
-	procedures, err := identityProviderPlan.Create(ctx)
 	if err != nil {
 		s.scope.Error(err, "failed creating eks identity provider plan")
 		return fmt.Errorf("creating eks identity provider plan: %w", err)
 	}
+
+	// nothing will be done, we can leave
+	if len(procedures) == 0 {
```

Comment:
This change here introduces a change in behaviour. Which might be fine, I'm just thinking aloud on this. Previously, we constantly reconciled the status even if it was the same. Which makes no sense. Why update it if the status is the same? I think this should be okay. Though I really would like to hear @dlipovetsky's opinion. :)

Reply:
ya i added this for exactly that reason... no point to update if everything is the same. i spent a lot of time in this code >.<